src/tools/clippy/clippy_utils/src/qualify_min_const_fn.rs

```diff
@@ -377,12 +377,12 @@ fn check_terminator<'tcx>(
     }
 }
 
-fn is_const_fn(tcx: TyCtxt<'_>, def_id: DefId, msrv: &Msrv) -> bool {
```

Comment:
@oli-obk FYI this renames `is_const_fn_raw` to `is_const_fn`, and replaces the previous `is_const_fn` by `is_stable_const_fn` which doesn't quite have the same semantics but close enough for the one (!) user in rustc that actually wants to use this.

A bunch of other places used `is_const_fn` when they shouldn't have -- in particular, [this one](https://github.com/rust-lang/rust/blob/d6f247f3d52f529e7145eb355ac9660a49d0d8c9/compiler/rustc_passes/src/reachable.rs#L47) is scary. Surely whether we consider something reachable should not depend on which feature gates are enabled in the current crate... Also, [this one](https://github.com/rust-lang/rust/blob/ab8673501ce13573c06b5989b179f5cfed85c771/compiler/rustc_mir_transform/src/check_undefined_transmutes.rs#L42) is supposed to lint in all const fn we declared, not just the ones for which we have the feature gate on.